SYCL/KernelAndProgram/cache-build-result.cpp

```diff
@@ -37,6 +37,11 @@ void test() {
         assert(Result == e.get_cl_code() && "Exception code differs");
       }
     } catch (...) {
+      try {
+        std::rethrow_exception(std::current_exception());
+      } catch (std::exception &E) {
+        fprintf(stderr, "Unexpected exception: %s\n", E.what());
+      }
       assert(false && "There must be cl::sycl::compile_program_error");
     }
```

Comment:
```suggestion
    } catch (std::exception &E) {
      fprintf(stderr, "Unexpected exception: %s\n", E.what());
    } catch (...) {
      assert(false && "There must be cl::sycl::compile_program_error");
    }
```
?

Reply:
We will never go to line 41-43 block. The assert may be moved to line 39-40 block

Reply:
>> We will never go to line 41-43 block. 

Could you please clarify why?

Reply:
sorry. my fault. I was thinking that all we throw is inherited from std::exception. It is definitely incorrect. So ignore my comment